kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -78,6 +79,59 @@ public static void validateSchema(StructType schema, boolean isColumnMappingEnab
     validateSupportedType(schema);
   }
 
+  public static void validateUpdatedSchema(StructType currentSchema, StructType schema) {
+    validateSchema(schema, true);
+    ColumnMapping.validateColumnIds(currentSchema, schema);
+    validateUpdatedSchemaCompatibility(currentSchema, schema);
+  }
+
+  private static void validateUpdatedSchemaCompatibility(
+      StructType currentSchema, StructType newSchema) {
+    // Identify added columns based on field IDs
+    Map<Integer, StructField> newSchemaIdToField = idToField(newSchema);
+    Map<Integer, StructField> currentSchemaIdToField = idToField(currentSchema);
+    for (Map.Entry<Integer, StructField> newFieldEntry : newSchemaIdToField.entrySet()) {
+      if (!currentSchemaIdToField.containsKey(newFieldEntry.getKey())) {
```

Comment:
@tdas validates that a non-nullable field cannot be added.

Reply:
Referencing, https://github.com/delta-io/delta/pull/4196#pullrequestreview-2651895034 @allisonport-db This already validates that a non-nullable field cannot be added.